ui/themes/materialplus/password.go

```diff
@@ -0,0 +1,297 @@
+
+	pinTabWidgets := &pinTabWidgets{
+		spendingEditor:         new(widget.Editor),
+		confirmEditor:          new(widget.Editor),
+		spendingEditorMaterial: t.Editor("Enter spending PIN"),
+		confirmEditorMaterial:  t.Editor("Enter spending PIN again"),
+	}
+
+	p := &PasswordAndPin{
+		cancelButton:         new(widget.Button),
+		cancelButtonMaterial: cancelButtonMaterial,
+
+		createButton:         new(widget.Button),
+		nextButton:           new(widget.Button),
+		nextButtonMaterial:   t.Button("Next"),
+		createButtonMaterial: t.Button("Create"),
+
+		passwordTabWidgets: passwordTabWidgets,
+		pinTabWidgets:      pinTabWidgets,
+		colors:             colors{},
+
+		errorLabel: errorLabel,
+	}
+
+	tabItems := []Tab{
+		{
+			ID:         passwordTabID,
+			Label:      "Password",
+			RenderFunc: p.passwordTab,
+		},
+		{
+			ID:         pinTabID,
+			Label:      "Pin",
+			RenderFunc: p.pinTab,
+		},
+	}
+	p.tabContainer = t.TabContainer(tabItems)
+
+	return p
+}
+
+func (p *PasswordAndPin) pinTab(gtx *layout.Context) {
+	w := []func(){
+		func() {
+			gtx.Constraints.Height.Min = 50
+			if !p.pinTabWidgets.isShowingConfirmEditor {
+				p.pinTabWidgets.spendingEditorMaterial.Layout(gtx, p.pinTabWidgets.spendingEditor)
+			} else {
+				p.pinTabWidgets.confirmEditorMaterial.Layout(gtx, p.pinTabWidgets.confirmEditor)
+			}
+		},
+	}
+
+	list := layout.List{Axis: layout.Vertical}
+	list.Layout(gtx, len(w), func(i int) {
+		inset := layout.Inset{
+			Top:   unit.Dp(7),
+			Left:  unit.Dp(25),
+			Right: unit.Dp(25),
+		}
+		inset.Layout(gtx, func() {
+			w[i]()
+		})
+	})
+}
+
+func (p *PasswordAndPin) passwordTab(gtx *layout.Context) {
+	w := []func(){
+		func() {
+			gtx.Constraints.Height.Min = 50
+			p.passwordTabWidgets.spendingEditorMaterial.Layout(gtx, p.passwordTabWidgets.spendingEditor)
+		},
+		func() {
+			gtx.Constraints.Height.Min = 50
+			p.passwordTabWidgets.confirmEditorMaterial.Layout(gtx, p.passwordTabWidgets.confirmEditor)
+		},
+	}
+
+	list := layout.List{Axis: layout.Vertical}
+	list.Layout(gtx, len(w), func(i int) {
+		inset := layout.Inset{
+			Top:   unit.Dp(7),
+			Left:  unit.Dp(25),
+			Right: unit.Dp(25),
+		}
+		inset.Layout(gtx, func() {
+			w[i]()
+		})
+	})
+}
+
+func (p *PasswordAndPin) updateColors() {
+	p.colors.cancelLabelColor = ui.GrayColor
+	p.colors.createButtonBackgroundColor = ui.GrayColor
+	p.colors.nextButtonBackgroundColor = ui.GrayColor
+
+	currentTabID := p.tabContainer.GetCurrentTabID()
+	p.colors.cancelLabelColor = ui.LightBlueColor
+
+	if currentTabID == pinTabID {
+		if p.pinTabWidgets.spendingEditor.Len() > 0 {
+			p.colors.nextButtonBackgroundColor = ui.LightBlueColor
+		}
+
+		if p.pinTabWidgets.isShowingConfirmEditor && p.bothPinsMatch() {
+			p.colors.createButtonBackgroundColor = ui.LightBlueColor
+		}
+	} else {
+		if p.bothPasswordsMatch() && p.passwordTabWidgets.confirmEditor.Len() > 0 {
+			p.colors.createButtonBackgroundColor = ui.LightBlueColor
+		}
+	}
+}
+
+// Draw renders this widget
```

Comment:
What are `createFunc` and `cancelFunc`? The argument names themselves give clues to their purpose but they have arguments themselves and that's where it gets confusing